internal/peermanager/peerwantmanager.go

```diff
@@ -79,26 +90,30 @@ func (pwm *peerWantManager) removePeer(p peer.ID) {
 // PrepareBroadcastWantHaves filters the list of want-haves for each peer,
 // returning a map of peers to the want-haves they have not yet been sent.
 func (pwm *peerWantManager) prepareBroadcastWantHaves(wantHaves []cid.Cid) map[peer.ID][]cid.Cid {
-	res := make(map[peer.ID][]cid.Cid)
+	res := make(map[peer.ID][]cid.Cid, len(pwm.peerWants))
```

Comment:
pre-allocate assuming we're likely to send _something_ to all peers. This map is short-lived anyways.